### PR TITLE
Submods data types, JSON and CBOR encoding/decoding

### PR DESCRIPTION
**Pull request by thomas-fossati**

### CODE REVIEW

File: eat_test.go [hunk 1]
```diff
@@ -51,11 +55,17 @@ var (
...
+
+	justEatSubmods = Eat{
+		Submods: &submods,
+	}
 )
 
 func cborRoundTripper(t *testing.T, tv Eat, expected []byte) {
 	data, err := tv.ToCBOR()
 
+	t.Logf("CBOR: %x", data)
```
Comment (setrofim) [inline eat_test.go:67]: Are these debug statements left in, or is the intention to leave them?

File: eat_test.go [hunk 2]
```diff
@@ -69,6 +79,8 @@ func cborRoundTripper(t *testing.T, tv Eat, expected []byte) {
 func jsonRoundTripper(t *testing.T, tv Eat, expected string) {
 	data, err := tv.ToJSON()
 
+	t.Logf("JSON: '%s'", string(data))
```
Comment (setrofim) [inline eat_test.go:82]: Are these debug statements left in, or is the intention to leave them?

Comment (thomas-fossati) [reply eat_test.go:82]: good question.  I'm actually undecided: they are quite useful when something goes bananas and they don't pollute stdout/stderr unless `go test` is invoked with the verbose flag.  So they are not super intrusive, although they are not nice to see...  I cannot convince myself one way or another.  You pick! :-)

Comment (setrofim) [reply eat_test.go:82]: If they're only visible with verbose flag, I don't see the harm in leaving them in.

**Review by setrofim — APPROVED** [review]

**Pull request merged**